app/monitoringapi.go

```diff
@@ -107,11 +112,31 @@ func startReadyChecker(ctx context.Context, tcpNode host.Host, eth2Cl eth2client
 	)
 	go func() {
 		ticker := clock.NewTicker(10 * time.Second)
+		epochTicker := clock.NewTicker(384 * time.Second) // 32 slots * 12 second slot time
+		current := make(map[core.PubKey]bool)
+		previous := make(map[core.PubKey]bool)
+
+		// Initialise current with given pubkeys.
+		for _, pubkey := range pubkeys {
+			current[pubkey] = true
+		}
 
 		for {
 			select {
 			case <-ctx.Done():
 				return
+			case <-epochTicker.Chan():
+				// Copy current to previous.
+				previous = make(map[core.PubKey]bool)
```

Comment:
```
// Copy current to previous and clear current
previous, current = current, newCurrent()
```

Reply:
this is incorrect, here previous will POINT to current.
https://go.dev/play/p/XsmNEqLRwNH

Reply:
no it is not: https://go.dev/play/p/9GruOUVMEHF

Reply:
yeah right, because current is also reinitialised